lib/jwt_sessions/store_adapters/abstract_store_adapter.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'redis'
+
+module JWTSessions
+  module StoreAdapters
+    def self.build_by_name(adapter, options = {})
+      camelized_adapter = adapter.to_s.split('_').map(&:capitalize).join
+      adapter_class_name = "JWTSessions::StoreAdapters::#{camelized_adapter}StoreAdapter"
+      Object.const_get(adapter_class_name).build(options)
+    end
+
+    class AbstractStoreAdapter
+      class << self
+        def build(token_prefix: JWTSessions.token_prefix, **store_options)
+          @_tokens_store = configured_store(store_options)
+          @_token_prefix = token_prefix || JWTSessions.token_prefix
+
+          new(@_tokens_store, @_token_prefix)
+        end
+
+        private :new
+      end
+
+      attr_reader :store, :prefix
+
+      def initialize(store, prefix)
+        @store  = store
+        @prefix = prefix
+      end
+
+      def fetch_access(uid)
+        csrf = store.get(access_key(uid))
+        return {} if csrf.nil?
+        { csrf: csrf }
+      end
+
+      def persist_access(uid, csrf, expiration)
+        key = access_key(uid)
+        store.set(key, csrf)
+        store.expireat(key, expiration)
+      end
+
+      REFRESH_KEYS = %i[csrf access_uid access_expiration expiration].freeze
+
+      def fetch_refresh(uid, namespace)
+        values = store.hmget(refresh_key(uid, namespace), *REFRESH_KEYS).compact
+        return {} if values.length != REFRESH_KEYS.length
+        REFRESH_KEYS.each_with_index.each_with_object({}) { |(key, index), acc| acc[key] = values[index]; }
+      end
+
+      def persist_refresh(uid, access_expiration, access_uid, csrf, expiration, namespace = '')
+        ns = namespace || ''
```

Comment:
Do we still need this `ns` local variable or we can use `namespace`?

Reply:
I've tried to do that but got `TestSession#test_flush_namespaced` failing, because this line makes sure we use an empty string instead of nil even if the nil was explicitly passed (test imitates this exact situation)